### PR TITLE
docs: add changelog for 4.2.3

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

docs

#### What this PR does / why we need it:

add changelog

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: CHANGELOG-zh-CN.md [hunk 1]
```diff
@@ -1,5 +1,108 @@
...
+
+- OceanBase MySQL 以及原生 MySQL 模式下支持 GIS 数据类型
+
+数据源
+
+- 允许数据源绑定到项目
+
+命令行窗口
+
+- 允许自定义禁用命令
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:21]: we not offer this feature to odc user, it is an obclient feature, odc just use it

File: CHANGELOG-zh-CN.md [hunk 2]
```diff
@@ -1,5 +1,108 @@
...
+
+- 任务占用过多内存
+- 不支持 ZHSGB232 编码
+- OceanBase MySQL 以及原生 MySQL 模式下无法对宽度在 8 以下的 bit 类型生成任务
+- 无法跳过自增的主键列
+- OceanBase MySQL 以及原生 MySQL 模式下 bit 类型宽度显示错误
+
+### 改进
+
+- 增加了在 SQL Check 以及任务预检查场景下的问题定位功能，允许用户将违反的安全规则和具体的 SQL 联系在一起
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:91]: it is new feature, all 'added' like should be new feature

File: CHANGELOG-zh-CN.md (hunk 1 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:21]: okay, delete it~

File: CHANGELOG-zh-CN.md (hunk 2 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:91]: done

**Review by smallsheeeep — COMMENTED** [review]
LGTM

File: CHANGELOG-zh-CN.md [hunk 3]
```diff
@@ -1,5 +1,146 @@
...
+
+- 支持原生 MySQL 数据源的导入导出
+- OceanBase 数据源的导入导出任务配置页面中不再提供 SYS 账户配置
+
+数据库对象管理
+
+- OceanBase MySQL 以及原生 MySQL 模式下支持 GIS 数据类型
+- 白屏创建或删除索引时给出高风险操作提示
+
+数据源
```
Comment (PeachThinking) [inline CHANGELOG-zh-CN.md:17]: Missing: Support for ODP sharding mysql data source

Comment (yhilmare) [reply CHANGELOG-zh-CN.md:17]: done

File: CHANGELOG-zh-CN.md [hunk 4]
```diff
@@ -1,5 +1,147 @@
...
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:17]: 1. move datasource category to the top.

2. ODC 4.2.3 new features may miss:
- SQL Diagnose: a lot of improvement for Full Chain TRACE.
- SQL Execution:  snippet support 100k maxium size,  support ctrl+enter shortcut for execute current statement.
- System ingegration: bastion integration can cowork with SQL Check  integraiton now.
- DLM: support change parameter, support view task log .
- Datasource: support clone operation, for fast create a new datasource

**Review by PeachThinking — APPROVED** [review]
LGTM

File: CHANGELOG-zh-CN.md [hunk 5]
```diff
@@ -1,5 +1,171 @@
...
+- 支持 Ctrl+Enter 快捷键执行当前语句
+
+堡垒机集成
+
+- 支持 SQL Check
+
+DLM
+
+- 支持日志查看
+
```
Comment (LuckyPickleZZ) [inline CHANGELOG-zh-CN.md:53]: This version of the FullLinkTrace has undergone these optimizations:
1. Able to output JSON files supported by Jaeger
2. Optimized visual effects
3. Added span list view, providing search and sorting functions for span

File: CHANGELOG-zh-CN.md [hunk 6]
```diff
@@ -1,5 +1,171 @@
...
+基于项目的管控协同
+
+- 增加 2 个内置项目角色：安全管理员，参与者；安全管理员被允许管理项目的敏感列和参与审批，参与者被允许参与审批
+- 允许用户申请项目权限
+- 禁止删除以 DBA 或 项目 OWNER 角色归属到任意项目中的用户
+
+SQL-Check
+
+- 增加了问题定位功能，允许用户将违反的安全规则和具体的 SQL 联系在一起
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:31]: suggestion: 
SQL 开发规范
- 优化了 SQL 拦截交互，同时支持快速定位原始 SQL 中的具体问题

File: CHANGELOG-zh-CN.md [hunk 7]
```diff
@@ -1,5 +1,171 @@
...
+外部审批集成
+
+- 无法识别索引集合中的数据的表达式
+- 外部系统返回的 xml 形式的数据在反序列化时会丢失原始 xml 的根 tag
+
+数据脱敏
+
+- 当扫描到重复列时会导致敏感列添加失败
+
+基于项目的管控协同
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:106]: suggestion: 基于项目的管控协同 ---> 项目

File: CHANGELOG-zh-CN.md [hunk 8]
```diff
@@ -1,5 +1,174 @@
 # OceanBase Developer Center (ODC) CHANGELOG
 
+## 4.2.3 (2023-12-21)
+
+### 功能变化
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:5]: ++
工单
- 支持查看所有工单。项目管理员能查看该项目下的所有工单，其它角色能查看自己审批过的工单

File: CHANGELOG-zh-CN.md [hunk 9]
```diff
@@ -1,5 +1,174 @@
...
+
+- 当扫描到重复列时会导致敏感列添加失败
+
+基于项目的管控协同
+
+- 用户被赋予"个人空间"权限后必须重新登陆才能生效
+- 同步大量数据库或 schema 到项目时发生事务超时
+- 无法以项目维度过滤工单
+- 项目 OWNER 可以将项目中所有 DBA 角色的用户全部移除
+- 项目参与者可以创建数据库并将其添加到项目中
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:112]: May delete this as the project participant is a new feature and this issue is introduced in 4.2.3

File: CHANGELOG-zh-CN.md [hunk 10]
```diff
@@ -1,5 +1,174 @@
...
+
+obclient 集成
+
+- 重复创建同名操作系统用户导致报错
+
+工单
+
+- 创建工单耗时过久
+
+操作审计
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:150]: 操作审计 ---> 操作记录

File: CHANGELOG-zh-CN.md [hunk 11]
```diff
@@ -1,5 +1,174 @@
...
+
+- 操作审计记录中"数据源"栏目为空
+- SQL 执行事件没有被记录
+- 打开 SQL 窗口事件没有被记录
+
+### 改进
```
Comment (MarkPotato777) [inline CHANGELOG-zh-CN.md:156]: ++ 支持展示团队空间对象树的数据源状态

File: CHANGELOG-zh-CN.md (hunk 6 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:31]: done

File: CHANGELOG-zh-CN.md (hunk 8 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:5]: done

File: CHANGELOG-zh-CN.md (hunk 9 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:112]: done

File: CHANGELOG-zh-CN.md (hunk 10 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:150]: done

File: CHANGELOG-zh-CN.md [hunk 12]
```diff
@@ -1,5 +1,174 @@
...
+
+- 提升 SQL 执行性能，减少不必要的耗时操作
+- 允许用户配置登录失败场景下的最大重试次数以及账户锁定时间
+- 不同步数据源内置的数据库或 schema
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:160]: this is only for 'binding datasource to project', I delete this

File: CHANGELOG-zh-CN.md (hunk 11 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:156]: done

File: CHANGELOG-zh-CN.md (hunk 7 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:118]: done

File: CHANGELOG-zh-CN.md (hunk 4 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md]: done

**Comment by krihy** [comment]
lgtm

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**